driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQDriver.java

```diff
@@ -0,0 +1,220 @@
+    private RabbitMQBrokerConfig rmqBrokerConfig;
+    private RabbitMQConfig rmqConfig;
+    private List<String> nodes;
+    private String user = "guest";
+    private String password = "guest";
+    private ConnectionFactory factory;
+    private RabbitMQChaosState state;
+    private GenericObjectPool<Channel> stateChannelPool;
+    private ObjectPool<Channel> producerChannelPool;
+    private GenericObjectPool<Channel> consumerChannelPool;
+    private Connection consumerConnection;
+    private Connection producerConnection;
+    private Connection stateConnection;
+    private Sync sync;
+    private volatile String curState = "stop";
+
+    private static RabbitMQClientConfig readConfigForClient(File configurationFile) throws IOException {
+        return MAPPER.readValue(configurationFile, RabbitMQClientConfig.class);
+    }
+
+    private static RabbitMQBrokerConfig readConfigForBroker(File configurationFile) throws IOException {
+        return MAPPER.readValue(configurationFile, RabbitMQBrokerConfig.class);
+    }
+
+    private static RabbitMQConfig readConfigForRMQ(File configurationFile) throws IOException {
+        return MAPPER.readValue(configurationFile, RabbitMQConfig.class);
+    }
+
+
+    @Override
+    public String getMetaNode() {
+        return null;
+    }
+
+    @Override
+    public String getMetaName() {
+        return null;
+    }
+
+    @Override
+    public void initialize(File configurationFile, List<String> nodes) throws IOException {
+        this.rmqClientConfig = readConfigForClient(configurationFile);
+        this.rmqBrokerConfig = readConfigForBroker(configurationFile);
+        this.rmqConfig = readConfigForRMQ(configurationFile);
+        this.nodes = nodes;
+        this.user = rmqClientConfig.user;
+        this.password = rmqClientConfig.password;
+        sync = new Sync(nodes);
+    }
+
+    public void intialState() {
+        if (Objects.equals("finish", curState) || Objects.equals("start", curState)) {
+            return;
+        }
+        curState = "start";
+        state = new RabbitMQChaosState("", queueName, nodes.get(0), rmqBrokerConfig.haMode, user, password);
+        ConnectionFactory tmpFac = new ConnectionFactory();
+        tmpFac.setHost(nodes.get(0));
+        tmpFac.setUsername(user);
+        tmpFac.setPassword(password);
+        try {
+            Connection tmpCon = tmpFac.newConnection("tmp");
+            Channel tmpChan = tmpCon.createChannel();
+            tmpChan.queueDelete(queueName);
+            tmpChan.queueDeclare(queueName, false, false, false, null);
+            tmpChan.close();
+            tmpCon.close();
+        } catch (IOException | TimeoutException e) {
+            throw new RuntimeException(e);
+        }
+        state = new RabbitMQChaosState("", queueName, state.getLeader().iterator().next(), rmqBrokerConfig.haMode, user, password);
+        factory = new ConnectionFactory();
+        factory.setHost(state.getLeader().iterator().next());
```

Comment:
If I didn't set address in host file for the leader rabbitmq node, an exception will be thrown here

Reply:
Thus users must set all address in host file before they run openchaos.